lib/node_modules/@stdlib/math/base/special/hyp2f1/lib/main.js

```diff
@@ -0,0 +1,285 @@
+* // returns 1.0
+*
+* @example
+* var v = hyp2f1( 10.0, 7.4, -1.8, -0.99 );
+* // returns ~0.423
+*
+* @example
+* var v = hyp2f1( 10.0, 1.0, -1.8, -0.99 );
+* // returns ~-0.875
+*
+* @example
+* var v = hyp2f1( 2.0, 3.0, 5.0, 0.99 );
+* // returns ~27.699
+*
+* @example
+* var v = hyp2f1( 3.0, 4.0, 7.0, 1.0 );
+* // returns +Infinity
+*
+* @example
+* var v = hyp2f1( NaN, 3.0, 2.0, 0.5 );
+* // returns NaN
+*
+* @example
+* var v = hyp2f1( 1.0, NaN, 2.0, 0.5 );
+* // returns NaN
+*/
+function hyp2f1( a, b, c, x ) {
+	var negIntCaOrCb;
+	var negIntC;
+	var negIntB;
+	var negIntA;
+	var isIntD;
+	var aid;
+	var err;
+	var ax;
+	var d2;
+	var d1;
+	var id;
+	var ic;
+	var ia;
+	var ib;
+	var t1;
+	var q;
+	var r;
+	var p;
+	var e;
+	var s;
+	var d;
+	var y;
+	var i;
+
+	err = 0.0;
+	ax = abs( x );
+	ia = round( a );
+	ib = round( b );
+	id = round( d );
+	ic = round( c );
+	s = 1.0 - x;
+	d = c - a - b;
+	negIntA = isNonPositiveInteger( a );
+	negIntB = isNonPositiveInteger( b );
+	negIntC = isNonPositiveInteger( c );
+	isIntD = isInteger( d );
+	t1 = abs( b - a );
+
+	if ( isnan( a ) || isnan( b ) || isnan( c ) || isnan( x ) ) {
+		return NaN;
+	}
+
+	if ( x === 0.0 ) {
+		return 1.0;
+	}
+
+	if ( ( a === 0.0 || b === 0.0 ) && c !== 0.0 ) {
+		return 1.0;
+	}
+
+	/* The transformation for c - a or c - b negative integer
+	* AMS55 #15.3.3
+	*/
+	if ( d <= -1.0 && !( !isIntD && s < 0.0 ) && !( negIntA || negIntB ) ) {
+		return pow( s, d ) * hyp2f1( c - a, c - b, c, x );
+	}
+
+	// The series diverges
+	if ( d <= 0.0 && x === 1.0 && !( negIntA || negIntB ) ) {
+		return PINF;
+	}
+
+	if ( ax < 1.0 || x === -1.0 ) {
+		if ( b === c ) {
+			// 2F1(a,b;b;x) = (1-x)**(-a)
+			return pow( s, -a );
+		}
+		if ( a === c ) {
+			// 2F1(a,b;a;x) = (1-x)**(-b)
+			return pow( s, -b );
+		}
+	}
+
+	if ( negIntC ) {
+		// Check if termination before explosion
+		if ( negIntA && ( ia > ic ) ) {
+			y = hyt2f1( a, b, c, x, err );
+			return y.value;
+		}
+		if ( negIntB && ( ib > ic ) ) {
+			y = hyt2f1( a, b, c, x, err );
+			return y.value;
+		}
+		return PINF;
+	}
+
+	// Function is a polynomial, so try the power series expansion
+	if ( negIntA || negIntB )	{
+		y = hyt2f1( a, b, c, x, err );
+		return y.value;
+	}
+
+	/**
+	* This transform has a pole for integer b - a and may cause large cancellation errors near |1/x| = 1.
+	* AMS55 #15.3.7
+	*/
+	if ( x < -2.0 && !isInteger( t1 ) ) {
+		p = hyp2f1( a, 1.0 - c + a, 1.0 - b + a, 1.0 / x );
+		q = hyp2f1( b, 1.0 - c + b, 1.0 - a + b, 1.0 / x );
+		p *= pow( -x, -a );
+		q *= pow( -x, -b );
+		t1 = gamma( c );
+		s = t1 * gamma( b - a ) / ( gamma( b ) * gamma( c - a ) );
+		y = t1 * gamma( a - b ) / ( gamma( a ) * gamma( c - b ) );
+		return ( s * p ) + ( y * q );
+	}
+	if ( x < -1.0 ) {
+		// Transformation based on AMS55 #15.3.4
+		if ( abs( a ) < abs( b ) ) {
+			return pow( s, -a ) * hyp2f1( a, c - b, c, x / ( x - 1.0 ) );
+		}
+		// Transformation based on AMS55 #15.3.5
```

Comment:
```suggestion
		// Transformation based on AMS55 #15.3.5:
```

Okay. General rules for line comments:

-   If the comment applies to the next 1-3 lines without conditionals, end with a `:`.
-   If the comment applies to a block of code (e.g., within an `if` block or a `for` loop, use `...`.

Why the distinction?

-   The former is intended to convey that a comment applies only to what immediately follows.
-   The latter is intended to convey that a comment applies more generally to a chunk of code as a whole.

This is a minor quibble, but to be consistent, it would be good to "clean" up the comments in the source implementation accordingly.

Reply:
On it!